packages/ubuntu_welcome/lib/services/xdg_session_service.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:meta/meta.dart';
+import 'package:ubuntu_desktop_installer/services/session_service.dart';
+import 'package:ubuntu_session/ubuntu_session.dart';
+
+class XdgSessionService implements SessionService {
+  XdgSessionService([@visibleForTesting UbuntuSession? session])
+      : _session = session ?? UbuntuSession();
+
+  final UbuntuSession _session;
+  @override
+  Future<void> reboot() {
+    return _session.reboot();
```

Comment:
does UbuntuSession have an argument that describes whether the reboot is "immediate" as subiquity calls it?

Reply:
No, this looks like some [subiquity-related parameter](https://github.com/canonical/subiquity/blob/fa29b9088bac7c55b9dc0588c63a18b18455fd78/subiquity/server/controllers/shutdown.py#L66-L72)

Reply:
the installer sends "immediate" reboot requests from the "turn off rst" and and "turn off bitlocker" pages. what if we used the systemd session manager that would allow passing interactive=false?

Reply:
According to the [man page](https://www.freedesktop.org/software/systemd/man/org.freedesktop.login1.html#Security):
> The interactive boolean parameters can be used to control whether polkit should interactively ask the user for authentication credentials if required.

So, setting `interactive=false` would just stop polkit from trying to get the user authentication to reboot/shutdown if necessary

Reply:
oh, not an interactive confirmation but an interactive authentication? the problem is that when called from the "turn off xxx" pages before installation, subiquity requires the immediate flag or else it won't reboot until installation is finished

Reply:
Ah, I see - if it's needed we could add the parameter and have it do nothing in the xdg case.